### PR TITLE
Deprecate authentication via a query string

### PR DESCRIPTION
**Pull request by zecakeh**
As per MSC4126.

It seems that OpenAPI 3.1 doesn't allow to mark a security scheme as deprecated, which is why I put it in the description.

Preview: https://pr1808--matrix-spec-previews.netlify.app

### CODE REVIEW

**Review by richvdh — APPROVED** [review]
Thanks!

File: changelogs/client_server/newsfragments/1808.clarification
Comment (Johennes) [inline changelogs/client_server/newsfragments/1808.clarification:1]: @zecakeh should this be a `.deprecation` as per the list in https://github.com/matrix-org/matrix-spec/blob/main/CONTRIBUTING.rst#adding-to-the-changelog?

Comment (turt2live) [reply changelogs/client_server/newsfragments/1808.clarification:1]: (we'll fix it during the changelog cleanup at release time - if there's other notable issues, please mention them in [#sct-office:matrix.org](https://matrix.to/#/#sct-office:matrix.org))

Comment (Johennes) [reply changelogs/client_server/newsfragments/1808.clarification:1]: Ok, yeah sorry for commenting on a closed PR. I wasn't sure what the proper process was. Would it help at all if I sent a PR renaming the file?

Comment (turt2live) [reply changelogs/client_server/newsfragments/1808.clarification:1]: Unfortunately not, as that PR then requires a changelog itself. Our [release process](https://github.com/matrix-org/matrix-spec/blob/main/meta/releasing.md) has a "normalize the changelog" step which takes care of this for us.

**Pull request merged**